### DISCUSSION TITLE
Proposition d'ajout de bonnes pratiques d'éco-conception d'application web

### DISCUSSION THREAD

**Discussion by okanfana** [cnumr/best-practices#436] [Bonnes pratiques]
Bonjour @ACTLEM, @rlemaire, @DocRoms , @tbroyer, 

Nous sommes heureux de vous contacter dans le but d'enréchir le référentiel des 115 bonnes pratiques en vous faisant part d'une  proposition d'ajout de nouvelles bonnes pratiques que nous avons obtenues à l'issue de nos recherches et analyses sur notre projet Green IT à Capgemini. En effet cette étude nous a permis d'identifier quatre bonnes pratiques d'éco-conception suivantes :

- Mettre à jour des logiciels que lorsque c’est indispensable
- Utiliser l’attribut d’auto-complétion de HTML pour de puissantes fonctions de remplissage automatique 
- Respectez le Cache Back/Forward
- Limitez les typographies

Nous proposerons d'ajouter ces quatre bonnes pratiques au référentiel. 
Qu'en pensez-vous ?
Merci

**Comment by tbroyer** [comment]
Pour la première, de quel logiciel parle-t-on exactement ?
Pour la seconde, si on parle de [l'attribut `autocomplete`](https://developer.mozilla.org/fr/docs/Web/HTML/Attributes/autocomplete), je ne suis pas sûr de voir la lien avec l'éco-conception.
Pour les 2 suivantes, elles sont respectivement [Respecter le principe de navigation rapide dans l’historique](https://github.com/cnumr/best-practices/blob/main/chapters/BP_007_fr.md) et [Favoriser les polices standards](https://github.com/cnumr/best-practices/blob/main/chapters/BP_029_fr.md)

**Comment by okanfana** [comment]
Bonjour, 
Nous faisons suite à votre dernier message

**Mise à jour de logiciels que lorsque celle-ci est indispensable :**
Les technologies évoluent sans cesse, chaque éditeur de logiciel cherche à perfectionner son programme de telle sorte qu’il soit le plus complet possible. Les mises à jour permettent donc d’ajouter de nouvelles fonctions aux logiciels et ainsi les rendre plus intéressant aux yeux des utilisateurs mais aussi plus goulu en consommation d'énergie et/ou de la mémoire. Etant donné qu’une mise à jour est une partie d’un programme qui est ajouté au programme existant, les logiciels ont tendance à prendre du poids avec l’âge.
Par exemple certaines mises à jour de windows bien que pour corriger des défauts, mais parfois il arrive que celles-ci soient mal optimisées et créent de nouveaux défauts au niveau soft ce qui ne rend  pas le parcours Utilisateur meilleur

** L’attribut d’auto-complétion :** cet attribut permet de mettre en place une assistance automatisée afin de remplir les valeurs des champs de formulaire ainsi que des indications au navigateur quant au type d'informations attendues, réduit le temps d'utilisation du site web par l'utilisateur (plus le temps d'utilisation est court moins d'énergie  consommée)

Après analyse, nous partagons votre point de vue qu'effectivement : le Cache Back/Forward, Limitez les typographies sont correspondantes à des pratiques déjà existantes du référentiel

**Reply by arnolem** [reply]
Bonjour,

Concernant la mise à jour de logiciel, je ne pense pas que l'on puisse écrire une règle là dessus.
En effet, une mise à jour peut également optimiser un programme et réduire son impact environnemental.

Concernant la réduction du temps passer sur un site, c'est probablement négligeable surtout que la majorité des sites web ne consomment plus rien côté serveur lorsque la page est chargée. La majorité des autocomplétions, bien au contraire, envoie de nombreuses requêtes au serveur à chaque caractère saisie.

**Reply by tbroyer** [reply]
@arnolem On ne parle pas de la même autocomplétion ici, on parle de celle que le navigateur est capable de faire ou assister avec les informations qu'il connait déjà (cf. [MDN](https://developer.mozilla.org/fr/docs/Web/HTML/Attributes/autocomplete) ou [cet article sur web.dev](https://web.dev/articles/payment-and-address-form-best-practices?hl=en))

Quant à la réduction du temps passé sur le site, on parle ici de l'impact du client (tel que pris en compte par exemple dans le simulateur de [Nos Gestes Climat](https://nosgestesclimat.fr/)), même si dans ce cas et à cette échelle c'est probablement à la marge et négligeable.
